src/main/java/fi/okm/jod/yksilo/service/KuvaService.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2024 The Finnish Ministry of Education and Culture, The Finnish
+ * The Ministry of Economic Affairs and Employment, The Finnish National Agency of
+ * Education (Opetushallitus) and The Finnish Development and Administration centre
+ * for ELY Centres and TE Offices (KEHA).
+ *
+ * Licensed under the EUPL-1.2-or-later.
+ */
+
+package fi.okm.jod.yksilo.service;
+
+import fi.okm.jod.yksilo.domain.JodUser;
+import fi.okm.jod.yksilo.entity.Kuva;
+import fi.okm.jod.yksilo.repository.KuvaRepository;
+import fi.okm.jod.yksilo.repository.YksiloRepository;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import net.coobird.thumbnailator.Thumbnails;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class KuvaService {
+  private final YksiloRepository yksilot;
+  private final KuvaRepository kuvat;
+
+  public Kuva add(JodUser user, MultipartFile file) throws IOException {
```

Comment:
It looks like a user can add unlimited number of pictures? Is that intended or should a new (profile) picture always replace the existing one?

Reply:
`@OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)` annotation in Yksilo entity ensures that there can only be one image at a time.